catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloadManager.java

```diff
@@ -123,58 +97,73 @@ public ResourceResponse download(ResourceRequest resourceRequest, Metacard metac
+            } else {
+                LOGGER.debug("Unable to get resource from cache. Have to retrieve it from source");
+            }
         }
 
-        resourceResponse.getProperties()
-                .put(Metacard.ID, metacard.getId());
-        // Sources do not create ResourceResponses with the original ResourceRequest, hence
-        // it is added here because it will be needed for caching
-        resourceResponse = new ResourceResponseImpl(resourceRequest,
-                resourceResponse.getProperties(),
-                resourceResponse.getResource());
-
-        // TODO - this should be before retrieveResource() but eventPublisher requires a
-        // resourceResponse and that resource response must have a resource request in it (to get
-        // USER property)
-        eventPublisher.postRetrievalStatus(resourceResponse,
-                ProductRetrievalStatus.STARTED,
-                metacard,
-                null,
-                0L,
-                downloadIdentifier);
-
-        AtomicBoolean downloadStarted = new AtomicBoolean(Boolean.FALSE);
-        ReliableResourceDownloader downloader = new ReliableResourceDownloader(downloaderConfig,
-                downloadStarted,
-                downloadIdentifier,
-                resourceResponse,
-                retriever);
-        resourceResponse = downloader.setupDownload(metacard, downloadStatusInfo);
-
-        // Start download in separate thread so can return ResourceResponse with
-        // ReliableResourceInputStream available for client to start reading from
-        executor.submit(downloader);
-
-        // Wait for download to get started before returning control to client
-        Stopwatch stopwatch = Stopwatch.createStarted();
-        while (!downloadStarted.get()) {
+        if (resourceResponse == null) {
             try {
-                Thread.sleep(10);
-            } catch (InterruptedException e) {
+                resourceResponse = retriever.retrieveResource();
+            } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+                throw new DownloadException("Cannot download resource", e);
             }
-            long elapsedTime = stopwatch.elapsed(TimeUnit.MILLISECONDS);
-            if (elapsedTime > ONE_SECOND_IN_MS) {
-                LOGGER.debug("downloadStarted still FALSE - elapsedTime = {}", elapsedTime);
-                break;
+
+            resourceResponse.getProperties()
+                    .put(Metacard.ID, metacard.getId());
+            // Sources do not create ResourceResponses with the original ResourceRequest, hence
+            // it is added here because it will be needed for caching
+            resourceResponse = new ResourceResponseImpl(resourceRequest,
+                    resourceResponse.getProperties(),
+                    resourceResponse.getResource());
+
+            // TODO - this should be before retrieveResource() but eventPublisher requires a
```

Comment:
I don't know if there is a ticket in place but we should probably create a ticket for it if there isn't already one. There will be more changes coming to the `ResourceCache` stuff soon though and it may get addressed there